pytorch_lightning/core/lightning.py

```diff
@@ -1697,24 +1701,13 @@ def summarize(self, mode: Optional[str] = "top", max_depth: Optional[int] = None
         Return:
             The model summary object
         """
-        model_summary = None
-
-        # temporary mapping from mode to max_depth
-        if max_depth is None:
-            if mode in ModelSummary.MODES:
-                max_depth = ModelSummary.MODES[mode]
-                rank_zero_deprecation(
-                    f"Argument `mode` in `LightningModule.summarize` is deprecated in v1.4"
-                    f" and will be removed in v1.6. Use `max_depth={max_depth}` to replicate `mode={mode}` behavior."
-                )
-                model_summary = ModelSummary(self, max_depth=max_depth)
-            elif mode is not None:
-                raise MisconfigurationException(f"`mode` can be None, {', '.join(ModelSummary.MODES)}, got {mode}")
-        else:
-            model_summary = ModelSummary(self, max_depth=max_depth)
+        warning_cache.deprecation(
+            "The `LightningModule.summarize` method is deprecated in v1.5 and will be removed in v1.7. "
+            "Use `pytorch_lightning.utilities.model_summary.summarize` instead.",
+            stacklevel=6,
+        )
 
-        log.info("\n" + str(model_summary))
-        return model_summary
+        return summarize(self, mode, max_depth)
```

Comment:
@awaelchli since we're moving the module anywas, what do you think about supporting `max_depth` only in the new utility function? then we don't have to support `mode` in the new spot until v1.6.

Reply:
We would only be able to do that if this PR lands for 1.4.

No strong opinion either way personally.

Reply:
We could do that since max_depth is only going to be introduced in 1.4. 
Waiting for @edenlightning's guidance on the milestones here :)